src/pass/store.rs

```diff
@@ -48,98 +46,107 @@ pub enum PasswordStoreError {
 #[derive(Debug, Deserialize, Serialize)]
 pub struct PasswordStore {
     passwords: Vec<PasswordEntry>,
+    master_password: MasterPassword<Verified>,
 }
 
 impl SerdeEncryptSharedKey for PasswordStore {
     type S = BincodeSerializer<Self>;
 }
 
 impl PasswordStore {
-    // Extract the data from database & store in PasswordStore
+    /// Extract the data from database(if exist) & store in PasswordStore
     pub fn new(
         file_path: impl AsRef<Path>,
-        master_password: impl AsRef<[u8]>,
+        master_password: MasterPassword<Verified>,
     ) -> Result<Self, PasswordStoreError> {
-        // If no PasswordEntry is stored yet then create file to store it
-        if !file_path.as_ref().exists() {
-            std::fs::File::create(file_path.as_ref())
-                .map_err(PasswordStoreError::UnableToCreateDirs)?;
-
-            // Returning an empty Vec bcs of no Entry available
-            return Ok(PasswordStore {
-                passwords: Vec::new(),
-            });
+        match file_path.as_ref().exists() {
+            // Load all existing password entry from db
+            true => PasswordStore::load(file_path.as_ref(), master_password),
+
+            // No password entry is stored yet, then create file to store it
+            false => {
+                std::fs::File::create(file_path.as_ref())
+                    .map_err(PasswordStoreError::UnableToCreateDirs)?;
```

Comment:
Misleading Error name, creating a file, error says Dirs

Reply:
Fixed it by making a new error `UnableToCreateFile`